backend/coding-standards.md

```diff
@@ -24,6 +24,32 @@ For our PHP projects we use [PHPCS](https://github.com/squizlabs/PHP_CodeSniffer
 
 We also use [PHPCD](https://github.com/sebastianbergmann/phpcpd) to ensure that code is not duplicated within the project.
 
+## AWS Resource Naming
+
+Historic resources don't follow an agreed convention. However, for all new resources, both in old and new projects, we should follow the schema:
+
+`${service-name}-${service-stage}-${resource-type}-${resource-label}`
```

Comment:
Is it worth using underscores as separators? Is this allowed for AWS resources? Just to differentiate between dashes in names

Also curious to know why it's useful to have the resource type in the name I suppose. Do all S3 buckets really need to contain S3 in the name?

Also would be tempted to change it to 

`${service-name}-${resource-label}-${service-stage}`

or

`${service-name}-${resource-type}-${resource-label}-${service-stage}`

Basically putting the label before the stage, so that the staging/production version of a given resource are grouped together when sorting by name, rather than grouping things by stage

Reply:
There are a few resources that don't support underscores.

What do everyone else think about having `stage` after `label`? To me there are two point to consider

1. We define a `prefix` in SLS so we can actual name things by using `${self:custom.prefix}` - which makes things easier to manage and to maintain
2. I really like having all staging and all production grouped together, so there can be no mix.

Reply:
Just a matter of taste I guess so I withdraw the suggestion. I more often want to look at the two versions of a given s3 bucket for example, rather than see all my staging buckets together - but there's always filtering!